elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,410 @@
+import graphql.language.Selection;
+import graphql.language.SelectionSet;
+import graphql.language.Value;
+import graphql.parser.Parser;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class converts a GraphQL query string into an Elide {@link EntityProjection} using {@link #make(String)}.
+ */
+@Slf4j
+public class GraphQLEntityProjectionMaker {
+    private final EntityDictionary entityDictionary;
+    private final ElideSettings elideSettings;
+
+    private final Map<String, List<EntityProjection>> fragmentToProjections = new HashMap<>();
+    private final Map<String, FragmentDefinition> fragmentDefinitions = new HashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param entityDictionary entityDictionary of current Elide instance
+     * @param elideSettings setting of current Elide instance
+     */
+    public GraphQLEntityProjectionMaker(EntityDictionary entityDictionary, ElideSettings elideSettings) {
+        this.entityDictionary = entityDictionary;
+        this.elideSettings = elideSettings;
+    }
+
+    private final List<EntityProjection> rootProjections = new ArrayList<>();
+
+    /**
+     * Convert a GraphQL query string into a collection of Elide {@link EntityProjection}s.
+     *
+     * @param query GraphQL query
+     * @return all projections in the query
+     */
+    public Collection<EntityProjection> make(String query) {
+        Parser parser = new Parser();
+        Document parsedDocument = parser.parseDocument(query);
+
+        parsedDocument.getDefinitions().forEach(definition -> {
+            if (definition instanceof OperationDefinition) {
+                // Operations would be converted into EntityProjection tree
+                OperationDefinition operationDefinition = (OperationDefinition) definition;
+                if (operationDefinition.getOperation() != OperationDefinition.Operation.QUERY) {
+                    // TODO: support MUTATION and SUBSCRIPTION
+                    return;
+                }
+
+                addRootProjections(operationDefinition.getSelectionSet());
+            } else if (definition instanceof FragmentDefinition) {
+                // FragmentDefinition should be stored first and then inserted into EntityProjections that use it
+                FragmentDefinition fragmentDefinition = (FragmentDefinition) definition;
+                fragmentDefinitions.put(fragmentDefinition.getName(), fragmentDefinition);
+            } else {
+                throw new InvalidEntityBodyException(
+                        String.format("Unsupported definition type {%s}.", definition.getClass()));
+            }
+        });
+
+        // make sure there is no fragment loop and undefined fragments in fragment definitions
```

Comment:
Let's break this up into more functions.  Validations can be separate functions.